db/cassandra/src/main/java/org/commonjava/indy/cassandra/data/CassandraStoreDataManager.java

```diff
@@ -0,0 +1,595 @@
+import static org.commonjava.indy.db.common.StoreUpdateAction.STORE;
+import static org.commonjava.indy.model.core.StoreType.group;
+
+@ApplicationScoped
+@Alternative
+public class CassandraStoreDataManager extends AbstractStoreDataManager
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    CassandraStoreQuery storeQuery;
+
+    @Inject
+    IndyObjectMapper objectMapper;
+
+    @Inject
+    @StoreByPkgCache
+    private CacheHandle<String, Map<StoreType, Set<StoreKey>>> storesByPkg;
+
+    @Inject
+    @AffectedByStoreCache
+    private CacheHandle<StoreKey, Set<StoreKey>> affectedByStores;
+
+    @Override
+    protected StoreEventDispatcher getStoreEventDispatcher()
+    {
+        return null;
+    }
+
+    @Override
+    protected ArtifactStore getArtifactStoreInternal( StoreKey key )
+    {
+        DtxArtifactStore dtxArtifactStore = storeQuery.getArtifactStore( key.getPackageType(), key.getType(), key.getName() );
+
+        return toArtifactStore( dtxArtifactStore );
+    }
+
+    @Override
+    protected ArtifactStore removeArtifactStoreInternal( StoreKey key )
+    {
+        DtxArtifactStore dtxArtifactStore = storeQuery.removeArtifactStore( key.getPackageType(), key.getType(), key.getName() );
+
+        final Map<StoreType, Set<StoreKey>> typedKeys = storesByPkg.get( key.getPackageType() );
+        if ( typedKeys != null )
+        {
+            final Set<StoreKey> keys = typedKeys.get( key.getType() );
+            if ( keys != null )
+            {
+                keys.remove( key );
+            }
+        }
+
+        return toArtifactStore( dtxArtifactStore );
+    }
+
+    @Override
+    public void clear( ChangeSummary summary ) throws IndyDataException
+    {
+
+    }
+
+    @Override
+    public Set<ArtifactStore> getAllArtifactStores()
+    {
+        Set<DtxArtifactStore> dtxArtifactStoreSet = storeQuery.getAllArtifactStores();
+        Set<ArtifactStore> artifactStoreSet = new HashSet<>(  );
+        dtxArtifactStoreSet.forEach( dtxArtifactStore -> {
+            artifactStoreSet.add( toArtifactStore( dtxArtifactStore ) );
+        } );
+        return artifactStoreSet;
+    }
+
+    @Override
+    public Map<StoreKey, ArtifactStore> getArtifactStoresByKey()
+    {
+        Map<StoreKey, ArtifactStore> ret = new HashMap<>();
+        Set<ArtifactStore> artifactStoreSet = getAllArtifactStores();
+        artifactStoreSet.forEach( store -> {
+            ret.put( store.getKey(), store );
+        } );
+        return ret;
+    }
+
+    @Override
+    public boolean hasArtifactStore( StoreKey key )
+    {
+        ArtifactStore artifactStore = getArtifactStoreInternal( key );
+        return artifactStore != null;
+    }
+
+    @Override
+    public boolean isStarted()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return false;
+    }
+
+    @Override
+    public Stream<StoreKey> streamArtifactStoreKeys()
```

Comment:
Are we implementing this? Can we remove it if we don't need it?

Reply:
Oh, good catch, we need to implement it.